packages/artifact/src/internal/requestUtils.ts

```diff
@@ -0,0 +1,74 @@
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
```

Comment:
I saw https://github.com/actions/toolkit/blob/main/packages/cache/src/internal/requestUtils.ts which you linked.  Would we be able to share the same basic retry logic?

Reply:
cc @dhadka and @joshmgross to see if we can share code with the cache action

Reply:
I think we'd need to move the shared code to it's own package, or add this all to https://github.com/actions/http-client

Reply:
A separate (small) NPM package or http-client would make sense. This feels like a long term investment though

Reply:
Ok, if it needs a separate package then I agree we can hold off.  We've tried sharing files among packages before in Azure Pipelines and it doesn't work well.

I think it could fit in in `@actions/io`, but that would mean making it a part of the public interface there.